abr_control/controllers/path_planners/path_planner.py

```diff
@@ -37,57 +480,258 @@ def convert_to_time(self, path, time_length):
 
         return path_func
 
-    def next(self):
-        """Returns the next target from the generated path"""
-        position = self.position_path[self.n]  # pylint: disable=E0203
-        velocity = self.velocity_path[self.n]  # pylint: disable=E0203
-        self.n = min(self.n + 1, self.n_timesteps - 1)
-        # some path planner may not end with zero target velocity depending on
-        # their parameters this will assure that you have zero target velocity
-        # when the filter is positionally at the final target
-        if self.n == self.n_timesteps - 1:
-            velocity = np.zeros(3)
-
-        return position, velocity
-
-    def _plot(self, target_position):
-        """Plot the generated trajectory
+    def _plot(self, start_position, target_position, ang=None):
```

Comment:
what is the `ang` parameter?

Reply:
Since you can optionally plan an orientation path, that parameter sets whether or not we're plotting orientaiton. However, most of the parameters used in this function are `self` variables set in `generate_path`. Since it's only internally called after running `generate_path`, I think it would make the most sense to remove all of the parameters from `_plot()` and just have `start_position` , `target_position` set as `self` variables. There is enough information in `generate_path` that we can get whether or not an orientation path exists and omit the `ang` parameter all together. I can make the changes if you push your latest, if you'd like.

Reply:
nooooo haha definitely it's preferred to explicitly pass things in as opposed to just using self variables. i've made a bunch of changes to this function, i'll push and then we can look at what that would be like

Reply:
The reason I did it this way is because the plotting function was only intended to be run internally when `plot=True` in `generate_path`. It uses variables that are created in `generate_path` that aren't passed back to the user, so I set it up with `self ` variables. But hey, you're Dr  Mr Bossman 👍